### PR TITLE
VGG19 encoder-decoder pairs using unpooling layers

### PR DESCRIPTION
**Pull request by emilutz**
I uploaded a set of encoder-decoders based on VGG19 with some modifications for better image reconstruction. One thing I didn't understand is how do models get their tfhub url when hosted? In the sample code I used a tfhub url for downloading the models which does not exist at the moment.

### CODE REVIEW

File: assets/docs/emilutz/collections/vgg19-unpooling-encoder-decoder/1.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
+# Collection emilutz/vgg19-unpooling-encoder-decoder/1
+
+Collection of image encoders and decoders trained on the COCO 2017 dataset where:
+
```
Comment (WGierke) [inline assets/docs/emilutz/collections/vgg19-unpooling-encoder-decoder/1.md:4]: Please remove this blank line

File: assets/docs/emilutz/models/vgg19-block1-conv2-unpooling-decoder/1.md [hunk 2]
```diff
@@ -0,0 +1,84 @@
+# Module emilutz/vgg19-block1-conv2-unpooling-decoder/1
```
Comment (WGierke) [inline assets/docs/emilutz/models/vgg19-block1-conv2-unpooling-decoder/1.md:1]: The handle in this line (and the location of the file in .../emilutz/models/vgg19-block1-conv2-unpooling-decoder/1.md) will make the model available at https://tfhub.dev/emilutz/vgg19-block1-conv2-unpooling-decoder/1 (so `tfhub.dev/HANDLE`). This doesn't match your code example i.e.
```python
decoder = hub.KerasLayer('https://tfhub.dev/vgg19-with-unpooling/vgg19-block1-conv2/decoder/1') # this model
```
I'd suggest that you update all the references to this model page to https://tfhub.dev/emilutz/vgg19-block1-conv2-unpooling-decoder/1.

File: assets/docs/emilutz/collections/vgg19-unpooling-encoder-decoder/1.md [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+  
+The training process consisted of 30 to 100 epochs on 90% of the data, while 9% was used for validation and 1% left for testing. Adam optimizer was used with the default parameters and a linear learning rate decay between 0.001 and 0.0002, and the only data augmentation used was a random horizontal flip on each batch.
+
+## Models
+
+Encoder inputs are expected to be batches of images of arbitrary height and width but depending on the encoder-decoder pair used VGG19-Block**X**-Conv2 the sizes must be multiples of 2 ^ **(X - 1)**. The outputs of the encoder are a list of the extracted image features + the argmax values for all maxpooling layers encountered until that point in the VGG19. For the decoders the input and output shapes are reversed.
+
+| Name | Input Shape | Output Shape | L1 loss* |
+|--|--|--|--|
+| [VGG19-Block1-Conv2 Unpooling Encoder](https://tfhub.dev/vgg19-with-unpooling/vgg19-block1-conv2/encoder/1) | (B, H, W, 3) | (B, H, W, 64) |   -   |
```
Comment (WGierke) [inline assets/docs/emilutz/collections/vgg19-unpooling-encoder-decoder/1.md:35]: As described below, I'd suggest using the following URL instead: https://tfhub.dev/emilutz/vgg19-block1-conv2-unpooling-decoder/1 (as well as for all the other referenced model pages)

File: assets/docs/emilutz/models/vgg19-block1-conv2-unpooling-decoder/1.md (hunk 2 above)
Comment (emilutz) [reply assets/docs/emilutz/models/vgg19-block1-conv2-unpooling-decoder/1.md:1]: Thanks, I was actually confused about this aspect. I updated all references to all models.

**Comment by WGierke** [comment]
Thank you for your contributions!

**Comment by emilutz** [comment]
Hi Willi

Thank you for reviewing my work and merging it. I found two issues with my
models after they went live on tfhub by running some tests.

1. One of them (
https://tfhub.dev/emilutz/vgg19-block4-conv2-unpooling-decoder/1) seems to
be using the assets of a different model (
https://tfhub.dev/emilutz/vgg19-block3-conv2-unpooling-decoder/1) . I
checked the paths in the .md files and the actual files on my storage space
and it all seems fine so I don't know exactly why this is happening.
2. The table on
https://tfhub.dev/emilutz/collections/vgg19-unpooling-encoder-decoder/1 is
rendered pretty badly. I am not a markdown expert so I think I might have
messed up the formatting, although on github it looks good. Any ideas on
this?

Many thanks,
Emil

**Comment by WGierke** [comment]
Hi Emil

Good catches, thank you for spotting them!
1) The correct assets should be live now. Would you mind checking whether the assets correspond to the right handles now?
2) I fixed that via https://github.com/tensorflow/tfhub.dev/pull/47 and the change is live: https://tfhub.dev/emilutz/collections/vgg19-unpooling-encoder-decoder/1

Thanks again for finding the issues!
Willi

**Comment by emilutz** [comment]
Hi Willi

Thank you for making the changes! I ran the tests and can confirm
everything works smoothly now.

Have a great day,
Emil

**Pull request merged**